From 11c33c350c04f31567bc761b55f2731e2aa3bb8f Mon Sep 17 00:00:00 2001
From: Lars Greiss <kufikugel@googlemail.com>
Date: Thu, 27 Mar 2014 05:54:15 +0100
Subject: [PATCH 028/201] Camera2: Don't report incorrect supported picture
 formats

If camera HAL does not return valid picture formats do not show wrong defaults

Currently it falls back to jpeg and raw to choose for the user. All our devices
 which do not have a proper
return of picture format only support jpg. So remove the entry to choose in
 this case completely

Change-Id: Ib4e4094ba0a57e7751f6568f6ec3c3f1dfc1ffb8
---
 src/com/android/camera/CameraSettings.java | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/com/android/camera/CameraSettings.java b/src/com/android/camera/CameraSettings.java
index 9074ded82..78564dee8 100755
--- a/src/com/android/camera/CameraSettings.java
+++ b/src/com/android/camera/CameraSettings.java
@@ -700,7 +700,7 @@ public class CameraSettings {
     private List<String> getSupportedPictureFormatLists() {
         String str = mParameters.get(KEY_QC_PICTURE_FORMAT);
         if (str == null) {
-            str = "jpeg,raw"; // if not set, fall back to default behavior
+            return null;
         }
         return split(str);
     }
-- 
2.17.1

